tests/org.eclipse.ui.workbench.texteditor.tests/src/org/eclipse/ui/workbench/texteditor/tests/OverlayAccess.java

```diff
@@ -0,0 +1,327 @@
+	}
+
+	@Override
+	public IFindReplaceTarget getTarget() {
+		return findReplaceLogic.getTarget();
+	}
+
+	private void restoreInitialConfiguration() {
+		find.setText("");
+		select(SearchOptions.GLOBAL);
+		unselect(SearchOptions.REGEX);
+		unselect(SearchOptions.CASE_SENSITIVE);
+		unselect(SearchOptions.WHOLE_WORD);
+	}
+
+	@Override
+	public void closeAndRestore() {
+		restoreInitialConfiguration();
+		assertInitialConfiguration();
+		closeOperation.run();
+	}
+
+	@Override
+	public void close() {
+		closeOperation.run();
+	}
+
+	@Override
+	public void select(SearchOptions option) {
+		ToolItem button= getButtonForSearchOption(option);
+		if (button == null) {
+			return;
+		}
+		button.setSelection(true);
+		if (option == SearchOptions.GLOBAL) {
+			button.setSelection(false);
+		}
+		button.notifyListeners(SWT.Selection, null);
+	}
+
+	@Override
+	public void unselect(SearchOptions option) {
+		ToolItem button= getButtonForSearchOption(option);
+		if (button == null) {
+			return;
+		}
+		button.setSelection(false);
+		if (option == SearchOptions.GLOBAL) {
+			button.setSelection(true);
+		}
+		button.notifyListeners(SWT.Selection, null);
+	}
+
+	@Override
+	public void simulateEnterInFindInputField(boolean shiftPressed) {
+		simulateKeyPressInFindInputField(SWT.CR, shiftPressed);
+	}
+
+	@Override
+	public void simulateKeyPressInFindInputField(int keyCode, boolean shiftPressed) {
+		final Event event= new Event();
+		event.type= SWT.KeyDown;
+		event.keyCode= keyCode;
+		if (shiftPressed) {
+			event.stateMask= SWT.SHIFT;
+		}
+		find.notifyListeners(SWT.KeyDown, event);
+		find.traverse(SWT.TRAVERSE_RETURN, event);
+		FindReplaceTestUtil.runEventQueue();
+	}
+
+	@Override
+	public String getFindText() {
+		return find.getText();
+	}
+
+	@Override
+	public String getReplaceText() {
+		return replace.getText();
+	}
+
+	@Override
+	public void setFindText(String text) {
+		find.setText(text);
+		find.notifyListeners(SWT.Modify, null);
+	}
+
+	@Override
+	public void setReplaceText(String text) {
+		openReplaceDialog();
+		replace.setText(text);
+	}
+
+	@Override
+	public ToolItem getButtonForSearchOption(SearchOptions option) {
+		switch (option) {
+			case CASE_SENSITIVE:
+				return caseSensitive;
+			case REGEX:
+				return regEx;
+			case WHOLE_WORD:
+				return wholeWord;
+			case GLOBAL:
+				return inSelection;
+			//$CASES-OMITTED$
+			default:
+				return null;
+		}
+	}
+
+	private Set<SearchOptions> getEnabledOptions() {
+		return Arrays.stream(SearchOptions.values())
+				.filter(option -> {
+					ToolItem buttonForSearchOption= getButtonForSearchOption(option);
+					if (option == SearchOptions.GLOBAL) {
+						return buttonForSearchOption.getEnabled();// The "Global" option is mapped to a button that
+						// selects whether to search in the selection, thus inverting the semantic
+					}
+					return buttonForSearchOption != null && buttonForSearchOption.getEnabled();
+				})
+				.collect(Collectors.toSet());
+	}
```

Comment:
This is a follow-up to https://github.com/eclipse-platform/eclipse.platform.ui/pull/1192#discussion_r1598370379 and the changes made to address it: why is there still a differentiation by the type of search option for checking the enablement? The special logic for `GLOBAL` implements the same behavior as for all other buttons and the comment for this is even incorrect.

Reply:
Since there is no differentiation anymore, the entire logic can be condensed like in the DialogAccess class.
The comment should have been updated, but it should not even be needed in the first place. I have removed it!

Reply:
https://github.com/eclipse-platform/eclipse.platform.ui/commit/2f9136a25069244769073ee77ad5be15b5dadeee